heat/core/logical.py

```diff
@@ -516,10 +516,13 @@ def sanitize_input_type(
     x = sanitize_input_type(x, y)
     y = sanitize_input_type(y, x)
 
-    # if one of the tensors is distributed, unsplit/gather it
-    if x.split is not None and y.split is None:
-        t1 = manipulations.resplit(x, axis=None)
-        return t1, y
+    # if one of the tensors is distributed and the other is not a scalar, unsplit/gather it
+    if x.split is not None and y.split is None and y.ndim > 0:
+        local_start = sum(x.lshape[: x.split])
+        local_end = local_start + x.lshape[x.split]
```

Comment:
The `lshape` attribute gives you information on the shape of the process-local data. `lshape` on a given process doesn't know about the size of slices of `x` distributed on other processes. 

Because you need to slice `y` along the distribution axis `x.split`, what you need here is information on how `x` is distributed across processes. I can think of 2 options:

**Option a):** use the information in `x.lshape_map`: 

`lshape_map` is 2-dimensional with shape (p, ndim) where p is the number of processes, ndim the number of dimension of a DNDarray. 
Example, say you have x.shape = (10, 5, 8) (this is the global shape) and x.split = 0, and say you're distributing across 3 MPI processes.  `x.lshape_map` might look like this
```
tensor([[4, 5, 8],
        [3, 5, 8],
        [3, 5, 8]])
```
The distribution along the first column (split axis = 0) might be different if `x` is not balanced.

**Option b):** use the `dndarray.counts_displs()` method which returns tuples of the counts (elements of `x` along the split axis for every process) and displacements (starting index of the local slice as referred to the global array). For the example above (shape (10, 5, 8) on 3 processes), `x.counts_displs()` will return 
```shell
>>> counts
(4, 3, 3)
>>> displs
(0,4,3)
```